### PR TITLE
Texture swap

### PR DESCRIPTION
**Pull request by RodPaDev**
This is working, haven't tested if saves get changed but I think they should. I will test it tomorrow

### CODE REVIEW

File: client/models/model_editor.js [hunk 1]
```diff
@@ -138,27 +138,55 @@ class Models {
...
     }
 
+    textureSelect(){
+        let select = document.getElementById("model-dialog-controls-texture");
+        select.options.length = 0;
+
+        let pack = this.loaded_asset_pack;
+
+        this.selected_type === 'asset'
+            ? createList(select, this.asset_packs[pack].textures)
```
Comment (Heleor) [inline client/models/model_editor.js]: Pass 'false' as the third argument to these lists -- 'delete' is not a valid texture name.

File: client/models/model_editor.js [hunk 2]
```diff
@@ -138,27 +138,55 @@ class Models {
...
+        
+        this.textureSelect();
         this.resetUI();
     }
 
+    textureSelect(){
+        let select = document.getElementById("model-dialog-controls-texture");
+        select.options.length = 0;
+
+        let pack = this.loaded_asset_pack;
```
Comment (Heleor) [inline client/models/model_editor.js]: Unnecessary variable

File: client/models/model_editor.js [hunk 3]
```diff
@@ -138,27 +138,55 @@ class Models {
...
+    }
+
+    textureChange(){
+        let selectVal = document.getElementById("model-dialog-controls-texture").value;
+        
+        let original = this.getCurrentModel();
+
+        this.selected_type === 'asset'
+            ? original.texture = selectVal
+            : original.texture = "models/shared-textures/" + selectVal;
```
Comment (Heleor) [inline client/models/model_editor.js]: a) Don't modify the current model, change the 'local_changes'.
b) I think it should be 'this.local_changes.sharedTexture = selectVal'

**Review by Heleor — CHANGES_REQUESTED** [review]

File: client/models/model_editor.js (hunk 1 above)
Comment (RodPaDev) [reply client/models/model_editor.js]: I passed True because the default for skip_delete is false

Comment (Heleor) [reply client/models/model_editor.js]: True is right. The original PR wasn't passing anything.

File: server/tools/scenery.js [hunk 4]
```diff
@@ -87,14 +87,54 @@ function deleteModel(workspace, body) {
     return true;
 }
 
-function createDefinition(workspace, body) {
-    console.log("Create Definition: " + workspace + ": " + json(body));
-    return false;
+function createDefinition(workspace, body) {
```
Comment (Heleor) [inline server/tools/scenery.js:90]: Can you rebase the PR so that it only shows the latest changes?

**Review by Heleor — CHANGES_REQUESTED** [review]

File: server/tools/scenery.js (hunk 4 above)
Comment (RodPaDev) [reply server/tools/scenery.js:90]: sure thing

**Pull request closed without merging**